### PR TITLE
Adding links for Atomics docs #29377

### PR DESCRIPTION
**Pull request by projektir**
r? @steveklabnik 

This should be good for `std::sync::atomic`. The other pages still need more (examples, etc.).

### CODE REVIEW

File: src/libcore/sync/atomic.rs [hunk 1]
```diff
@@ -15,27 +15,37 @@
...
-//! Atomic variables are safe to share between threads (they implement `Sync`)
+//! Atomic variables are safe to share between threads (they implement [`Sync`])
 //! but they do not themselves provide the mechanism for sharing and follow the
 //! [threading model](../../../std/thread/index.html#the-threading-model) of rust.
-//! The most common way to share an atomic variable is to put it into an `Arc` (an
+//! The most common way to share an atomic variable is to put it into an [`Arc`][arc] (an
 //! atomically-reference-counted shared pointer).
 //!
+//! [`Sync`]: ../../marker/trait.Sync.html
+//! [arc]: ../struct.Arc.html
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: https://travis-ci.org/rust-lang/rust/jobs/215789767#L8942 (It says this link is broken.)

The reason for this is that these docs _also_ get rendered for `libcore`, not just `libstd`. So can you change it to `../../../std/sync/struct.Arc.html`? Tricky!

File: src/libcore/sync/atomic.rs [hunk 2]
```diff
@@ -149,21 +159,26 @@ unsafe impl<T> Sync for AtomicPtr<T> {}
 #[derive(Copy, Clone, Debug)]
 pub enum Ordering {
     /// No ordering constraints, only atomic operations. Corresponds to LLVM's
-    /// `Monotonic` ordering.
+    /// [`Monotonic`][1] ordering.
+    /// [1]: http://llvm.org/docs/Atomics.html#monotonic
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: While you're here, would you mind helping with a minor formatting issue as well? This should look like this:

```rust
     /// No ordering constraints, only atomic operations.
     ///
     /// Corresponds to LLVM's [`Monotonic`] ordering.
     ///
     /// [`Monotonic`]: http://llvm.org/docs/Atomics.html#monotonic
```

Thanks 😄

File: src/libcore/sync/atomic.rs [hunk 3]
```diff
@@ -149,21 +159,26 @@ unsafe impl<T> Sync for AtomicPtr<T> {}
...
     #[stable(feature = "rust1", since = "1.0.0")]
     Relaxed,
     /// When coupled with a store, all previous writes become visible
-    /// to the other threads that perform a load with `Acquire` ordering
+    /// to the other threads that perform a load with [`Acquire`][1] ordering
     /// on the same value.
+    /// [1]: http://llvm.org/docs/Atomics.html#acquire
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: Similar here, could this be

```rust
    /// to the other threads that perform a load with [`Acquire`] ordering
    /// on the same value.
    ///
    /// [`Aquire`]: http://llvm.org/docs/Atomics.html#acquire
```

File: src/libcore/sync/atomic.rs [hunk 4]
```diff
@@ -149,21 +159,26 @@ unsafe impl<T> Sync for AtomicPtr<T> {}
...
     #[stable(feature = "rust1", since = "1.0.0")]
     Release,
     /// When coupled with a load, all subsequent loads will see data
-    /// written before a store with `Release` ordering on the same value
+    /// written before a store with [`Release`][1] ordering on the same value
     /// in other threads.
+    /// [1]: http://llvm.org/docs/Atomics.html#release
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: same here

File: src/libcore/sync/atomic.rs [hunk 5]
```diff
@@ -149,21 +159,26 @@ unsafe impl<T> Sync for AtomicPtr<T> {}
...
     #[stable(feature = "rust1", since = "1.0.0")]
     Acquire,
-    /// When coupled with a load, uses `Acquire` ordering, and with a store
-    /// `Release` ordering.
+    /// When coupled with a load, uses [`Acquire`][1] ordering, and with a store
+    /// [`Release`][2] ordering.
+    /// [1]: http://llvm.org/docs/Atomics.html#acquire
+    /// [2]: http://llvm.org/docs/Atomics.html#release
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: and here:

```rust
    /// When coupled with a load, uses [`Acquire`] ordering, and with a store
    /// [`Release`] ordering.
    /// 
    /// [`Acquire`]: http://llvm.org/docs/Atomics.html#acquire
    /// [`Release`]: http://llvm.org/docs/Atomics.html#release
```

File: src/libcore/sync/atomic.rs [hunk 6]
```diff
@@ -176,7 +191,8 @@ pub enum Ordering {
     __Nonexhaustive,
 }
 
-/// An `AtomicBool` initialized to `false`.
+/// An [`AtomicBool`] initialized to `false`.
+/// [`AtomicBool`]: struct.AtomicBool.html
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: an extra `///` between these two lines, please!

File: src/libcore/sync/atomic.rs [hunk 7]
```diff
@@ -241,7 +257,7 @@ impl AtomicBool {
 
     /// Loads a value from the bool.
     ///
-    /// `load` takes an [`Ordering`] argument which describes the memory ordering
+    /// `load()` takes an [`Ordering`] argument which describes the memory ordering
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: we don't include the `()` in function names; some of our docs do, but we decided against that style.

File: src/libcore/sync/atomic.rs [hunk 8]
```diff
@@ -269,7 +285,7 @@ impl AtomicBool {
 
     /// Stores a value into the bool.
     ///
-    /// `store` takes an [`Ordering`] argument which describes the memory ordering
+    /// `store()` takes an [`Ordering`] argument which describes the memory ordering
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: same here

File: src/libcore/sync/atomic.rs [hunk 9]
```diff
@@ -250,7 +266,7 @@ impl AtomicBool {
     ///
     /// [`Ordering`]: enum.Ordering.html
     /// [`Release`]: enum.Ordering.html#variant.Release
-    /// [`AcqRel`]: enum.Ordering.html#variant.Release
+    /// [`AcqRel`]: enum.Ordering.html#variant.AcqRel
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: nice catch! 😓

File: src/libcore/sync/atomic.rs [hunk 10]
```diff
@@ -298,7 +317,7 @@ impl AtomicBool {
 
     /// Stores a value into the bool, returning the old value.
     ///
-    /// `swap` takes an [`Ordering`] argument which describes the memory ordering
+    /// `swap()` takes an [`Ordering`] argument which describes the memory ordering
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: same here

File: src/libcore/sync/atomic.rs [hunk 11]
```diff
@@ -324,7 +343,7 @@ impl AtomicBool {
     /// The return value is always the previous value. If it is equal to `current`, then the value
     /// was updated.
     ///
-    /// `compare_and_swap` also takes an [`Ordering`] argument which describes the memory
+    /// `compare_and_swap()` also takes an [`Ordering`] argument which describes the memory
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: and here (I'll stop now, but all the others too)

File: src/libcore/sync/atomic.rs [hunk 12]
```diff
@@ -404,17 +423,18 @@ impl AtomicBool {
...
     /// previous value.
     ///
-    /// `compare_exchange_weak` takes two [`Ordering`] arguments to describe the memory
+    /// `compare_exchange_weak()` takes two [`Ordering`] arguments to describe the memory
     /// ordering of this operation. The first describes the required ordering if the operation
     /// succeeds while the second describes the required ordering when the operation fails. The
     /// failure ordering can't be [`Release`] or [`AcqRel`] and must be equivalent or
     /// weaker than the success ordering.
     ///
+    /// [`compare_exchange()`]: #method.compare_exchange
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: when removing the `()`s this one will have to go too

File: src/libcore/sync/atomic.rs [hunk 13]
```diff
@@ -1008,7 +1032,11 @@ macro_rules! atomic_int {
             ///
             /// # Panics
             ///
-            /// Panics if `order` is `Acquire` or `AcqRel`.
+            /// Panics if `order` is [`Acquire`] or [`AcqRel`].
+            ///
+            /// [`Acquire`]: enum.Ordering.html#variant.Acquire
+            /// [`AcqRel`]: enum.Ordering.html#variant.AcqRel
+            ///
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: could you remove this extra line please?

**Review by steveklabnik — CHANGES_REQUESTED** [review]
Thank you so much! I have some small nits to make the formatting fall in line with some of our conventions, and one bugfix. 🎊

File: src/libcore/sync/atomic.rs (hunk 1 above)
Comment (projektir) [inline src/libcore/sync/atomic.rs]: Yeah, I had some issues with this one. I wasn't sure if a longer path was better for this to get linked in both place correctly. I'll play around it some more.

File: src/libcore/sync/atomic.rs (hunk 7 above)
Comment (projektir) [inline src/libcore/sync/atomic.rs]: Good to know! I'll be sure to fix this up in the Once PR, as well.

File: src/libcore/sync/atomic.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: For now it is, yeah. Someday...

**Comment by projektir** [comment]
@steveklabnik I've pushed a new commit for the formatting changes.

**Review by steveklabnik — APPROVED** [review]
@bors: r+ rollup

thanks so much!

**Comment by steveklabnik** [comment]
Oops, looks like @bors doesn't see review comments.

@bors: r+ rollup

**Pull request merged**